hibernate-core/src/main/java/org/hibernate/collection/spi/PersistentCollection.java

```diff
@@ -213,7 +216,7 @@ public interface PersistentCollection<E> extends LazyInitializable {
 	 *
 	 * @return The snapshot
 	 */
-	Serializable getSnapshot(CollectionPersister persister);
+	Serializable getSnapshot(@Nullable CollectionPersister persister);
```

Comment:
```suggestion
	Serializable getSnapshot(CollectionPersister persister);
```

Reply:
I had to add a castNonNull on a callsite to be able to remove it, because this is one of these cases where CF doesn't infer nullability correctly, not even by reversing the condition on the call site.